tests/unit/services/test_project.py

```diff
@@ -92,6 +92,9 @@ def test_add_pebble_part_noble(
     [
         ("bare", "devel"),
         ("ubuntu@25.10", "devel"),
+        # base is not allowed to be empty, but checking nonetheless since
+        # pebble part is added pre-YAML validation
```

Comment:
Could you add a negative test for this as well (that if there's no pebble part and both base and build_base are None, it doesn't add the pebble part)?